### ISSUE TITLE
Can't use base classes as parameters

### ISSUE THREAD

**Issue by rdking** [microsoft/TypeScript#52843]
# Bug Report
### 🔎 Search Terms
jsdoc
class
parameter
validation

### 🕗 Version & Regression Information

I ran into this in VSCode 1.75.1. I don't know what version of TypeScript that corresponds to.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/dev/bug-workbench/?#code/PQKhCgAIUgVALAlgZ0gd0QG05A9gE30wE9IBjAQwFdkBTSCyABwCdcAjTWgW3XloB2kGrXwA6KCGDgAZlQFkALolxDFtZIoAUAa0wVkyAFyRFxJrVwzIAMXlKVAgJSQA3gF9w4dZq1l9hpAAogAeFNxMXACMbu5OANxePtr+Bqih4ZG0AEyQtCHqAviodgrKqrEJ4EA)

### 💻 Code

```ts
/**
 * This will oddly cause a problem when used.
 */
function test(klass: typeof Function) {}

test(class Example1 {});
```

### 🙁 Actual behavior

Compiler Errors
Argument of type 'typeof Example1' is not assignable to parameter of type 'FunctionConstructor'.
  Types of property 'prototype' are incompatible.
    Type 'Example1' is missing the following properties from type 'Function': apply, call, bind, prototype, and 5 more.

Example 1 has every one of the properties of Function as that is the super class of every class by default. So the error makes no sense.

### 🙂 Expected behavior

This should not give any error at all and simply work.

**Comment by fatcerberus** [comment]
All classes are `Function`, but *not* `typeof Function`, which is the type of the [`Function` constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/Function) itself.  In other words it's the type you get for `x` if you write `const x = Function;`  No class is a subclass of that.

**Comment by rdking** [comment]
That's problematic. So what's the correct way to say you're expecting `klass` to be a constructor function? I thought that was the purpose of `typeof`.

**Comment by fatcerberus** [comment]
No, the purpose of `typeof` is to get the type of an identifier in scope.  So `typeof x` would give you the type of the local variable `x`.  It’s not specifically the “class constructor” operator.

You can just use `Function` in this case.

**Comment by rdking** [comment]
That doesn't work. If test is defined as such:
```ts
function test(klass: typeof Function) {
   new klass();
}

test(class Example1 {});
```
... then I get:

> Compiler Errors
>   This expression is not constructable.
>     Type 'Function' has no construct signatures.

What I want is to be able to pass in an instantiate-able function. There doesn't seem to be a way to do this that I can find.

**Comment by fatcerberus** [comment]
```ts
function test<T>(klass: new () => T): T {
   return new klass();
}

test(class Example1 {});
```

**Comment by rdking** [comment]
Ok. I get it. It's just far more cumbersome than I would have liked, but it's there. 

Thanks for the help.

**Comment by fatcerberus** [comment]
Alternatively, you could use `new () => any`, but I wouldn't recommend that.  That's why I wrote a generic version.
